src/Components/InMemoryEventStream.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Components
+{
+    internal sealed class InMemoryEventStream : EventStream, EventDispatcher
+    {
+        private readonly HashSet<object> _subscriptions = new HashSet<object>();
+
+        public void Dispatch<T>(T entity)
+        {
+            if (entity == null)
+                throw new ArgumentNullException(nameof(entity));
+
+            lock(_subscriptions)
```

Comment:
Would we be better served have a concurrent collection? (is there a reason you didn't use one?)